File: 11-2DArrays/Solution.py
#!/bin/python3

import sys


arr = []
for arr_i in range(6):
   arr_t = [int(arr_temp) for arr_temp in input().strip().split(' ')]
   arr.append(arr_t)
    
result = []

for i in range(4):
    for j in range(4):
        hsum = arr[i][j] + arr[i][j+1] + arr[i][j+2] \
        + arr[i+1][j+1] \
        + arr[i+2][j] + arr[i+2][j+1] + arr[i+2][j+2]
        result.append(hsum)
        
print(max(result))

File: 11-2DArrays/Solution.cpp
#include <vector>
#include <climits>
#include <iostream>

using namespace std;


int main(){
    vector< vector<int> > arr(6,vector<int>(6));
    for(int arr_i = 0;arr_i < 6;arr_i++){
       for(int arr_j = 0;arr_j < 6;arr_j++){
          cin >> arr[arr_i][arr_j];
       }
    }
    
    int maximum=INT_MIN;
    int hsum = 0;
        
    
    for (int i = 0; i < 4; i++) {
        for (int j = 0; j < 4; j++) {
            hsum = arr[i][j] + arr[i][j+1] + arr[i][j+2]
                + arr[i+1][j+1]
                + arr[i+2][j] + arr[i+2][j+1] + arr[i+2][j+2];
            
            if (maximum < hsum) {
                maximum = hsum;
            }            
        }
    }
    
    cout << maximum;
    
    return 0;
}

